src/Jackett.Common/Indexers/HDBitsApi.cs

```diff
@@ -119,6 +119,16 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                 }
             }
 
+            if (configData.Origins.Values.Length > 0)
+            {
+                requestData["origin"] = new JArray();
+
+                foreach (var origin in configData.Origins.Values)
+                {
+                    requestData["origin"].Add(new JValue(int.Parse(origin)));
+                }
+            }
+
```

Comment:
This follows the pattern of the surrounding code, but I'm not a fan of the dynamic type identifier since we miss out on the compiler static checking and code completion. Also, all these options can be added much more simply. As example, this code could be:
```C#
if(configData.Origins.Values.Any())
    requestData.add("origin", JToken.FromObject(configData.Origins.Values.Select(int.Parse));
```

At some point, this tracker should be refactored accordingly.

Reply:
Sure, will take a look at it at some point tomorrow!